src/Microsoft.AspNet.SignalR.SqlServer/DbOperation.cs

```diff
@@ -180,43 +168,28 @@ private void LoggerCommand(DbCommand command)
         [SuppressMessage("Microsoft.Usage", "CA2202:Do not dispose objects multiple times", Justification = "Disposed in async Finally block"),
          SuppressMessage("Microsoft.Reliability", "CA2000:Dispose objects before losing scope", Justification = "Disposed in async Finally block")]
 #if ASPNET50
-        private void Execute<T>(Func<IDbCommand, Task<T>> commandFunc, TaskCompletionSource<T> tcs)
+        private async void Execute<T>(Func<IDbCommand, Task<T>> commandFunc, TaskCompletionSource<T> tcs)
 #else
-        private void Execute<T>(Func<DbCommand, Task<T>> commandFunc, TaskCompletionSource<T> tcs)
+        private async void Execute<T>(Func<DbCommand, Task<T>> commandFunc, TaskCompletionSource<T> tcs)
 #endif
         {
-#if ASPNET50
-            IDbConnection connection = null;
-#else
-            DbConnection connection = null;
-#endif
-            try
+            using (var connection = _dbProviderFactory.CreateConnection())
             {
-                connection = _dbProviderFactory.CreateConnection();
                 connection.ConnectionString = ConnectionString;
                 var command = CreateCommand(connection);
 
                 connection.Open();
```

Comment:
this should be inside try for sure. if `CreateCommand` can throw it should be inside the try block as well.